ballerina/http_client_endpoint.bal

```diff
@@ -658,20 +658,21 @@ isolated function performDataBinding(Response response, TargetType targetType) r
         return response.getTextPayload();
     } else if (targetType is typedesc<string?>) {
         string|ClientError payload = response.getTextPayload();
-        return payload is ClientError ? () : payload;
+        return payload is NoContentError ? () : payload;
     } else if (targetType is typedesc<xml>) {
         return response.getXmlPayload();
     } else if (targetType is typedesc<xml?>) {
         xml|ClientError payload = response.getXmlPayload();
-        return payload is ClientError ? () : payload;
+        return payload is NoContentError ? () : payload;
     } else if (targetType is typedesc<byte[]>) {
         return response.getBinaryPayload();
     } else if (targetType is typedesc<byte[]?>) {
         byte[]|ClientError payload = response.getBinaryPayload();
         if payload is byte[] {
             return payload.length() == 0 ? () : payload;
+        } else {
+            return payload is NoContentError ? () : payload;
```

Comment:
Does `response.getBinaryPayload();` return `NoContentError`? Can you check that?

Reply:
As per the code, it will return an empty array if the payload is nil, hence we do not need this additional check. Updated in [6332e64f743929c84d18df29f3132629b76bd4f2](https://github.com/ballerina-platform/module-ballerina-http/pull/612/commits/6332e64f743929c84d18df29f3132629b76bd4f2).